src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -199,21 +199,22 @@ ss::future<log_recovery_result> partition_downloader::download_log() {
         throw;
     } catch (const ss::gate_closed_exception&) {
         throw;
-    } catch (...) {
-        // We can get here if the parttion manifest is missing (or some
-        // other failure is preventing us from recovering the partition). In
+    } catch (const missing_partition_exception& err) {
+        // We can get here in case if manifest doesn't exist in S3. In
         // this case the exception can't be propagated since the partition
         // manager will retry and it will create an infinite loop.
         //
         // The only possible solution here is to discard the exception and
         // continue with normal partition creation process.
-        //
-        // Normally, this will happen when one of the partitions doesn't
-        // have any data.
+        vlog(_ctxlog.error, "Error during log recovery: {}", err);
```

Comment:
Is this covered by a test yet? Would be great to confirm that we actually don't upload anything, or better yet that there aren't any seriously broken invariants here.

It's unfortunate that there's no read-only mode for partitions or somesuch. IIUC, to a user, this partition will seem perfectly functional until it's too late

Reply:
Also, could you explicitly construct and return a log_recovery_result? It looks surprising (and will probably look confusing in the future) that we're just fall through.